sycl/test/extensions/annotated_usm/annotated_usm.cpp

```diff
@@ -0,0 +1,243 @@
+  free(APtr53, q);
+  free(APtr6, q);
+  free(APtr7, q);
+}
+
+void TestProperty(sycl::queue &q) {
+  // P4: input properties
+  properties P4{conduit, cache_config{large_slm}};
+  // ANNP4: output properties, with no usm_kind and no runtime property
+  properties ANNP4{conduit};
+  auto APtr1 = malloc_annotated<int>(N, q, alloc::device, P4);
+  static_assert(
+      std::is_same_v<decltype(APtr1), annotated_ptr<int, decltype(ANNP4)>>);
+
+  // APtr7 is of type annotated_ptr<int, decltype(properties{})>;
+  auto APtr2 = malloc_annotated<int>(N, q, alloc::device);
+  static_assert(std::is_same_v<decltype(APtr2), annotated_ptr<int>>);
+
+  properties P5{usm_kind<alloc::host>};
+  // Throws an exception with error code errc::invalid
+  // auto APtr8 = malloc_annotated<int>(N, q, sycl::usm::alloc::device, P5);
+
+  // Error: the USM kinds do not agree
+  // auto APtr9 = malloc_shared_annotated<int>(N, q, P5);
+
+  // the USM kind is required in the propList
+  auto APtr3 = malloc_annotated<int>(N, q, P5);
+  static_assert(
+      std::is_same_v<decltype(APtr3), annotated_ptr<int, decltype(P5)>>);
+
+  auto APtr31 = malloc_annotated(N, q, P5);
+  static_assert(
+      std::is_same_v<decltype(APtr31), annotated_ptr<void, decltype(P5)>>);
+
+  free(APtr1, q);
+  free(APtr2, q);
+  free(APtr3, q);
+  free(APtr31, q);
+}
+
+void TestAlign(sycl::queue &q) {
+  // P7: input properties
+  properties P7{alignment<512>};
+  // ANNP7: output properties with usm_kind
+  properties ANNP7{alignment<512>, usm_kind<alloc::device>};
+  // The raw pointer of APtr1 is 512-byte aligned
+  auto APtr1 = malloc_device_annotated<int>(N, q, P7);
+  static_assert(
+      std::is_same_v<decltype(APtr1), annotated_ptr<int, decltype(ANNP7)>>);
+
+  // P8: input properties
+  properties P8{alignment<2>};
+  // ANNP8: output properties with usm_kind
+  properties ANNP8{alignment<2>, usm_kind<alloc::device>};
+  // The raw pointer of APtr2 is sizeof(int)-byte aligned, e.g., 4 for some
+  // implementations
+  auto APtr2 = malloc_device_annotated<int>(N, q, P8);
+  static_assert(
+      std::is_same_v<decltype(APtr2), annotated_ptr<int, decltype(ANNP8)>>);
+
+  // P9: input properties
+  properties P9{alignment<64>};
+  // ANNP9: output properties with usm_kind
+  properties ANNP9{alignment<64>, usm_kind<alloc::device>};
+  // The raw pointer of APtr3 is 64-byte aligned
+  auto APtr3 = malloc_device_annotated(N, q, P9);
+  static_assert(
+      std::is_same_v<decltype(APtr3), annotated_ptr<void, decltype(ANNP9)>>);
+
+  // The raw pointer of APtr4 is 1023-byte aligned
+  // Note: APtr4 does not have the alignment
+  // property. The alignment is runtime information.
+  auto APtr4 = aligned_alloc_device_annotated<int>(
+      1024 /* alignment */, N, q,
+      sycl::ext::oneapi::experimental::detail::empty_properties_t{});
+  static_assert(
+      std::is_same_v<
+          decltype(APtr4),
+          annotated_ptr<int, decltype(properties{usm_kind<alloc::device>})>>);
```

Comment:
For consistency and to factor out shared code, can you define a properties object like:
`properties ANNP10{usm_kind<alloc::device>};`

ANNP10 could then be used here and in the next test case as well.